src/manifolds/MetricManifold.jl

```diff
@@ -51,6 +51,127 @@ inner product ``g(X, X) > 0`` whenever ``X`` is not the zero vector.
 """
 abstract type RiemannianMetric <: AbstractMetric end
 
+@doc raw"""
+    change_gradient(M::AbstractManifold, G2::AbstractMetric, p, X)
+
+Convert the gradient `X` at `p` on the[`AbstractManifold`](@ref) `M` from one metric to another.
+
+Assume that for a real-valued function ``f: \mathcal M \to ℝ`` the input `X` is the gradient or in
+other words the [Riesz representer](https://en.wikipedia.org/wiki/Riesz_representation_theorem#Riesz_representation_theorem) of the differential ``Df(p)``` with respect to the metric ``g_2`` i.e.
+
+```math
+    g_2(X,Y) = Df(p)[Y] \quad \text{for all } Y ∈ T_p\mathcal M.
+```
+
+(It could actually also be the Riesz representer of _any_ linear function, not just ``Df(p)``).
+
+In order to convert this into the gradient with respect to the (implicitly given) metric ``g_1`` of `M`,
+we have to find the conversion function ``c: T_p\mathcal M \to \T_p\mathcal M`` such that
+
+```math
+    g_2(X,Y) = g_1(c(X),Y)
+```
```

Comment:
Technically the whole discussion about gradient isn't essential to this function, and what it does is described by the formula in line 72. Alternatively it could be described as changing the metric with respect to which the Riesz representer of a cotangent vector is taken.

Reply:
Exactly! Thats exactly why I am not sure about the name yet. Do you have a good idea for the name of the function then?

Reply:
Maybe something like `representer_in_metric`?

Reply:
That misses the action of changing the representer. So I was thinking of maybe `change_representer`? The metric ideas would get clear from the signature (because otherwise the name would be too long)?

Reply:
Sure, `change_representer` sounds better and the metric part is indeed clear from the signature :+1: .

Reply:
Perfect. I will change the name and the docs accordingly. I think the more general name is really better.